backend/src/main/java/eu/clarin/switchboard/resources/DataResource.java

```diff
@@ -183,19 +194,55 @@ public Response getOutline(@Context HttpServletRequest request, @PathParam("id")
         if (fi == null) {
             return Response.status(Response.Status.NOT_FOUND).build();
         }
-        if (!fi.getProfile().toProfile().isMediaType("application/zip")) {
-            LOGGER.debug("getOutline returns: no content");
-            return Response.status(Response.Status.NO_CONTENT).build();
+
+        if (fi.getProfile().toProfile().isMediaType(Constants.MEDIATYPE_ZIP)) {
+            try {
+                Outline outline = extractOutlineFromZip(fi.getPath().toFile());
+                return Response.ok(outline).build();
+            } catch (ZipException xc) {
+                LOGGER.info("bad zip archive: " + xc.getMessage());
+                return Response.status(Response.Status.NOT_ACCEPTABLE).entity(xc.getMessage()).build();
+            }
+        } else if (fi.getProfile().toProfile().isMediaType(Constants.MEDIATYPE_TAR)) {
+            Outline outline = extractOutlineFromTar(fi.getPath().toFile());
+            return Response.ok(outline).build();
         }
 
-        ZipFile zfile = new ZipFile(fi.getPath().toFile());
-        List<ZipEntry> ret = zfile.stream()
-                .filter(e -> !e.isDirectory() && e.getSize() > 0)
-                .filter(e -> !e.getName().startsWith("__MACOSX/"))
-                .map(e -> new ZipEntry(e.getName(), e.getSize()))
-                .sorted(Comparator.comparing(ZipEntry::getName))
-                .collect(Collectors.toList());
-        return Response.ok(ret).build();
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+
+    private static Outline extractOutlineFromZip(File zipfile) throws IOException {
+        try (ZipFile zfile = new ZipFile(zipfile)) {
+            List<ZEntry> outline = zfile.stream()
+                    .filter(e -> !e.isDirectory() && e.getSize() > 0)
+                    .filter(e -> !e.getName().startsWith("__MACOSX/"))
```

Comment:
I think it is OK for now but I find this OSX specific filter very arbitrary. Perhaps there should be a place (class?) where we define the filter rules about what to show or not.

Reply:
Agree, this was moved into the next PR, should be visible here: https://github.com/clarin-eric/switchboard/pull/206/commits/7c28046b0ae55065b09112e005c6ad407800686e#diff-7c9d6e6d1582338bd85c5a4ea194309d11b404b62e8eedf28d7e220d7bb77084R25